plugins/backstage-plugin-coder/docs/guides/coder-api.md

```diff
@@ -0,0 +1,257 @@
+# Coder API - Quick-start guide
+
+## Overview
+
+The Coder plugin makes it easy to bring the entire Coder API into your Backstage deployment. This guide covers how to get it set up so that you can start accessing Coder from Backstage.
+
+Note: this covers the main expected use cases with the plugin. For more information and options on customizing your Backstage deployment further, see our [Advanced API guide](./coder-api-advanced.md).
+
+### Before you begin
+
+Please ensure that you have the Coder plugin fully installed before proceeding. You can find instructions for getting up and running in [our main README](../../README.md).
+
+### Important hooks for using the Coder API
+
+There are a few React hooks that are needed to interact with the Coder API. These can be split into three categories: React Query hooks, core plugin hooks, and convenience hooks.
+
+#### React Query hooks
+
+The Coder plugin uses [React Query/TanStack Query v4](https://tanstack.com/query/v4/docs/framework/react/overview) for all of its data caching. We recommend that you use it for your own data caching, because of the sheer amount of headaches it can spare you.
+
+There are three main hooks that you will likely need:
+
+- `useQuery` - Query and cache data
+- `useMutation` - Perform mutations on an API resource
+- `useQueryClient` - Coordinate queries and mutations
```

Comment:
Seems like basic use cases for fetching don't need these RQ hooks but they're the first thing we talk about.
WDYT about moving your example code snippets up to the top here alongside your hook explanations so users can immediately see how they might use what you've built?

This *could* look something like:
```
### Available hooks
We've exposed a few hooks to get you started building your own Coder components: 

- `useCoderApi` - Exposes an object with all available Coder API methods. For the most part, there is no exposed state on this object; you can consider it a "function bucket".
- `useCoderAuth` - Provides methods and state values for interacting with your current Coder auth session from within Backstage.
- `useCoderQuery` - a convenience hook that simplifies wiring up `useQuery`, `useCoderApi`, and `useCoderAuth`.

// Query Example:


// Mutation Example:

Note that we use [React Query/TanStack Query v4](https://tanstack.com/query/v4/docs/framework/react/overview) in the above examples. When querying, RQ's `useQuery` hook is also a good escape hatch if `useCoderQuery` doesn't meet your needs, but it requires more work to wire up correctly. 
The plugin exposes a `CODER_QUERY_KEY_PREFIX` constant that you can use to group all Coder queries together; read more here(link to your query constant section).

Reply:
Yeah, I see what you mean. I went ahead and reorganized the hooks section, and added some bite-sized examples